gimp-with-plugins does not compile anymore

## Issue description

I can't install anything anymore on my system because `gimp-with-plugins` fails to install, and when I run `nixos-rebuild switch`, I get an error during the compilation. Note that you can obtain it directly from `nix-build -E "with import <nixpkgs> {}; gimp-with-plugins"`

```
# nixos-rebuid switch
[...]
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:57:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimppixelrgn.h:48:11: note: declared here
 void      gimp_pixel_rgn_init       (GimpPixelRgn  *pr,
           ^~~~~~~~~~~~~~~~~~~
src/gimplensfun.cpp:923:77: warning: 'void gimp_pixel_rgn_get_rect(GimpPixelRgn*, guchar*, gint, gint, gint, gint)' is deprecated: Use 'gegl_buffer_get' instead [-Wdeprecated-declarations]
     gimp_pixel_rgn_get_rect (&rgn_in, ImgBuffer, x1, y1, imgwidth, imgheight);
                                                                             ^
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:57:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimppixelrgn.h:80:11: note: declared here
 void      gimp_pixel_rgn_get_rect   (GimpPixelRgn  *pr,
           ^~~~~~~~~~~~~~~~~~~~~~~
src/gimplensfun.cpp:949:35: warning: 'static lfModifier* lfModifier::Create(const lfLens*, float, int, int)' is deprecated [-Wdeprecated-declarations]
     lfModifier *mod = lfModifier::Create (lenses[0], sLensfunParameters.Crop, imgwidth, imgheight);
-- Found KF5ItemModels: /nix/store/0xgz2s87px14r74kmbqhaimjrwv0yn96-kitemmodels-5.58.0-dev/lib/cmake/KF5ItemModels/KF5ItemModelsConfig.cmake (found version "5.58.0")
                                   ^~~~~~
In file included from src/gimplensfun.cpp:30:0:
/nix/store/sslsjphx656mg6sawyixmm6fd61nfmg5-lensfun-0.3.95/include/lensfun/lensfun.h:2402:35: note: declared here
     DEPRECATED static lfModifier *Create (const lfLens *lens, float crop, int width, int height);
                                   ^~~~~~
src/gimplensfun.cpp:949:98: warning: 'static lfModifier* lfModifier::Create(const lfLens*, float, int, int)' is deprecated [-Wdeprecated-declarations]
     lfModifier *mod = lfModifier::Create (lenses[0], sLensfunParameters.Crop, imgwidth, imgheight);
                                                                                                  ^
In file included from src/gimplensfun.cpp:30:0:
/nix/store/sslsjphx656mg6sawyixmm6fd61nfmg5-lensfun-0.3.95/include/lensfun/lensfun.h:2402:35: note: declared here
     DEPRECATED static lfModifier *Create (const lfLens *lens, float crop, int width, int height);
                                   ^~~~~~
src/gimplensfun.cpp:952:84: warning: 'int lfModifier::Initialize(const lfLens*, lfPixelFormat, float, float, float, float, lfLensType, int, bool)' is deprecated [-Wdeprecated-declarations]
                          sLensfunParameters.ModifyFlags, sLensfunParameters.Inverse);
                                                                                    ^
In file included from src/gimplensfun.cpp:30:0:
/nix/store/sslsjphx656mg6sawyixmm6fd61nfmg5-lensfun-0.3.95/include/lensfun/lensfun.h:2449:20: note: declared here
     DEPRECATED int Initialize (
                    ^~~~~~~~~~
src/gimplensfun.cpp:1010:81: warning: 'void gimp_pixel_rgn_set_rect(GimpPixelRgn*, const guchar*, gint, gint, gint, gint)' is deprecated: Use 'gegl_buffer_set' instead [-Wdeprecated-declarations]
     gimp_pixel_rgn_set_rect (&rgn_out, ImgBufferOut, x1, y1, imgwidth, imgheight);
                                                                                 ^
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:57:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimppixelrgn.h:104:11: note: declared here
 void      gimp_pixel_rgn_set_rect   (GimpPixelRgn  *pr,
           ^~~~~~~~~~~~~~~~~~~~~~~
src/gimplensfun.cpp:1012:34: warning: 'void gimp_drawable_flush(GimpDrawable*)' is deprecated: Use 'gegl_buffer_flush' instead [-Wdeprecated-declarations]
     gimp_drawable_flush (drawable);
                                  ^
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:41:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimpdrawable.h:55:16: note: declared here
 void           gimp_drawable_flush                  (GimpDrawable  *drawable);
                ^~~~~~~~~~~~~~~~~~~
src/gimplensfun.cpp:1018:35: warning: 'void gimp_drawable_detach(GimpDrawable*)' is deprecated [-Wdeprecated-declarations]
     gimp_drawable_detach (drawable);
                                   ^
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:41:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimpdrawable.h:53:16: note: declared here
 void           gimp_drawable_detach                 (GimpDrawable  *drawable);
                ^~~~~~~~~~~~~~~~~~~~
src/gimplensfun.cpp: In function 'int read_opts_from_exif(const char*)':
src/gimplensfun.cpp:1063:19: error: 'AnyError' in namespace 'Exiv2' does not name a type
     catch (Exiv2::AnyError& e) {
                   ^~~~~~~~
src/gimplensfun.cpp: In function 'void run(const gchar*, gint, const GimpParam*, gint*, GimpParam**)':
src/gimplensfun.cpp:1224:59: warning: 'GimpDrawable* gimp_drawable_get(gint32)' is deprecated: Use 'gimp_drawable_get_buffer' instead [-Wdeprecated-declarations]
     drawable = gimp_drawable_get (param[2].data.d_drawable);
                                                           ^
In file included from /nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimp.h:41:0,
                 from src/gimplensfun.cpp:31:
/nix/store/89n53mna0flc0h8grbdagfawdmqyim0g-gimp-2.10.12/include/gimp-2.0/libgimp/gimpdrawable.h:51:16: note: declared here
 GimpDrawable * gimp_drawable_get                    (gint32         drawable_ID);
                ^~~~~~~~~~~~~~~~~
src/gimplensfun.cpp:1269:18: warning: 'void lfDatabase::Destroy()' is deprecated [-Wdeprecated-declarations]
     ldb->Destroy();
                  ^
In file included from src/gimplensfun.cpp:30:0:
/nix/store/sslsjphx656mg6sawyixmm6fd61nfmg5-lensfun-0.3.95/include/lensfun/lensfun.h:1757:21: note: declared here
     DEPRECATED void Destroy ();
                     ^~~~~~~
make: *** [Makefile:51: src/gimplensfun.o] Error 1
builder for '/nix/store/xpxgy5hxdgn2g043cx2xl2lcyq31ic7z-gimp-plugin-gimplensfun-0.2.4.drv' failed with exit code 2
cannot build derivation '/nix/store/s9lly3h1rmj71rmv5l1qfvsj6d6mcp0c-gimp-with-plugins-2.10.12.drv': 1 dependencies couldn't be built
copying path '/nix/store/sgcc7zkpajshmgq5051rxxfmssib9bhl-python-2.7.16-env' from 'https://cache.nixos.org'...
building '/nix/store/46dkiwk8w5hlia53ysx92si2fa9jvp0j-initrd-fsinfo.drv'...
building '/nix/store/mza8rgw0g0rc7df16aqqh49abj2i3whm-keymap.drv'...
cannot build derivation '/nix/store/rw2nhjy95jq2apsjxvi2k7r2vk89z6a0-system-path.drv': 1 dependencies couldn't be built
building '/nix/store/x7ni38bpn2p1j2x71kdlml870rslwks9-udev-path.drv'...
building '/nix/store/yi87vyi1mwmv9s0mx7rln38wcby2j2qz-udev-rules.drv'...
building '/nix/store/pc6v5pqxpgq0n2y2n4kia69md419ismb-unit-NetworkManager-dispatcher.service.drv'...
cannot build derivation '/nix/store/3ib2cmqh6ljvlbjpncrv0kac5zv0jl8q-nixos-system-colnixos-19.09pre184855.beff2f8d75e.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3ib2cmqh6ljvlbjpncrv0kac5zv0jl8q-nixos-system-colnixos-19.09pre184855.beff2f8d75e.drv' failed
```

### Steps to reproduce

Install or compile `gimp-with-plugins`:

    $ nix-build -E "with import <nixpkgs> {}; gimp-with-plugins"

I am on `19.09pre184855.beff2f8d75e`.

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.52, NixOS, 19.09pre183392.83ba5afcc96 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(me): `""`
 - channels(root): `"nixos-19.09pre184855.beff2f8d75e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/applications/graphics/gimp/plugins/default.nix
+++ b/pkgs/applications/graphics/gimp/plugins/default.nix
@@ -157,17 +157,17 @@ stdenv.lib.makeScope pkgs.newScope (self: with self; {
   ufraw = pkgs.ufraw.gimpPlugin;
 
   gimplensfun = pluginDerivation rec {
-    version = "0.2.4";
+    version = "unstable-2018-10-21";
     name = "gimplensfun-${version}";
 
     src = fetchFromGitHub {
       owner = "seebk";
       repo = "GIMP-Lensfun";
-      rev = version;
-      sha256 = "0zlmp9v732qmzj083mnk5z421s57mnckmpjhiw890wmmwzj2lhxz";
+      rev = "1c5a5c1534b5faf098b7441f8840d22835592f17";
+      sha256 = "1jj3n7spkjc63aipwdqsvq9gi07w13bb1v8iqzvxwzld2kxa3c8w";
     };
 
-    buildInputs = with pkgs; [ lensfun exiv2 ];
+    buildInputs = with pkgs; [ lensfun gexiv2 ];
 
     installPhase = "
       installPlugins gimp-lensfun

--- a/pkgs/applications/graphics/ufraw/default.nix
+++ b/pkgs/applications/graphics/ufraw/default.nix
@@ -1,38 +1,71 @@
-{ fetchurl, stdenv, pkgconfig, gtk2, gettext, bzip2, zlib
-, withGimpPlugin ? true, gimp ? null
-, libjpeg, libtiff, cfitsio, exiv2, lcms2, gtkimageview, lensfun }:
+{ stdenv
+, fetchFromGitHub
+
+, autoconf
+, automake
+, autoreconfHook
+, bzip2
+, cfitsio
+, exiv2
+, gettext
+, gimp ? null
+, gtk2
+, gtkimageview
+, lcms2
+, lensfun
+, libjpeg
+, libtiff
+, perl
+, pkgconfig
+, zlib
+
+, withGimpPlugin ? true
+}:
 
 assert withGimpPlugin -> gimp != null;
 
 stdenv.mkDerivation rec {
-  name = "ufraw-0.22";
+  pname = "ufraw";
+  version = "unstable-2019-06-12";
 
-  src = fetchurl {
-    # XXX: These guys appear to mutate uploaded tarballs!
-    url = "mirror://sourceforge/ufraw/${name}.tar.gz";
-    sha256 = "0pm216pg0vr44gwz9vcvq3fsf8r5iayljhf5nis2mnw7wn6d5azp";
+  # The original ufraw repo is unmaintained and broken;
+  # this is a fork that collects patches
+  src = fetchFromGitHub {
+    owner = "sergiomb2";
+    repo = "ufraw";
+    rev = "c65b4237dcb430fb274e4778afaf5df9a18e04e6";
+    sha256 = "02icn67bsinvgliy62qa6v7gmwgp2sh15jvm8iiz3c7g1h74f0b7";
   };
 
   outputs = [ "out" ] ++ stdenv.lib.optional withGimpPlugin "gimpPlugin";
 
-  nativeBuildInputs = [ pkgconfig gettext ];
+  nativeBuildInputs = [ autoconf automake autoreconfHook gettext perl pkgconfig ];
+
   buildInputs = [
-    gtk2 gtkimageview bzip2 zlib
-    libjpeg libtiff cfitsio exiv2 lcms2 lensfun
+    bzip2
+    cfitsio
+    exiv2
+    gtk2
+    gtkimageview
+    lcms2
+    lensfun
+    libjpeg
+    libtiff
+    zlib
   ] ++ stdenv.lib.optional withGimpPlugin gimp;
 
   configureFlags = [
-    "--enable-extras"
-    "--enable-dst-correction"
     "--enable-contrast"
+    "--enable-dst-correction"
+    "--enable-extras"
   ] ++ stdenv.lib.optional withGimpPlugin "--with-gimp";
 
   postInstall = stdenv.lib.optionalString withGimpPlugin ''
     moveToOutput "lib/gimp" "$gimpPlugin"
   '';
 
-  meta = {
-    homepage = http://ufraw.sourceforge.net/;
+  meta = with stdenv.lib; {
+    homepage = https://github.com/sergiomb2/ufraw;
 
     description = "Utility to read and manipulate raw images from digital cameras";
 
@@ -46,9 +79,9 @@ stdenv.mkDerivation rec {
          the camera's tone curves.
       '';
 
-    license = stdenv.lib.licenses.gpl2Plus;
+    license = licenses.gpl2Plus;
 
-    maintainers = [ ];
-    platforms = stdenv.lib.platforms.gnu ++ stdenv.lib.platforms.linux;  # needs GTK+
+    maintainers = with maintainers; [ gloaming ];
+    platforms   = with platforms; all;
   };
 }

--- a/pkgs/applications/graphics/ufraw/default.nix
+++ b/pkgs/applications/graphics/ufraw/default.nix
@@ -57,7 +57,6 @@ stdenv.mkDerivation rec {
   configureFlags = [
     "--enable-contrast"
     "--enable-dst-correction"
-    "--enable-extras"
   ] ++ stdenv.lib.optional withGimpPlugin "--with-gimp";
 
   postInstall = stdenv.lib.optionalString withGimpPlugin ''

